uk.co.compucorp.civicrm.hrcontactactionsmenu/js/contactactions.js

```diff
@@ -1,11 +1,52 @@
-(function ($) {
-  $(document)
-    // Actions menu
-    .on('click', function(e) {
-      if ($(e.target).is('#crm-contact-actions-link, #crm-contact-actions-link *')) {
-        $('#crm-contact-actions-link').addClass('active');
-      } else {
-        $('#crm-contact-actions-link').removeClass('active');
-      }
+(function (CRM) {
+  var $ = CRM.$;
+
+  (function init () {
+    initListeners();
+  }());
+
+  /**
+   * Displays confirmation dialog before deleting user account
+   */
+  function confirmUserAccountDeletion () {
+    var url = $('[data-delete-user-url]').attr('data-delete-user-url');
+
+    CRM.confirm({
+      'title': 'Confirmation dialog',
+      'message': 'Are you sure you want to continue?'
+    })
+    .on('crmConfirm:yes', function () {
+      window.location = url;
     });
-}(CRM.$));
+  }
+
+  /**
+   * Initializes click listeners for
+   * - Actions Button
+   * - Delete User Account Button
+   */
+  function initListeners () {
+    $(document).on('click', function (e) {
```

Comment:
Please include the listener inside the function itself, it will help to keep `initListeners` to be more clean and the code to be more organised.
```
function toggleActionMenuButtonClass () {
   $(document).on('click', function (e) { //listener code}
}
```

same for `confirmUserAccountDeletion `

Reply:
I see you have agreed to the solution, but it is not implemented. Could you check again please?